File: TP01/Palindromo.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

#define N 2000
int testaPalindromo(char palavra[]);
int TesteFim(char palavra[]);

int main()
{
    char palavra[N];

    do
    {
        scanf("%[^\n]", palavra);


        if (testaPalindromo(palavra) == 1)
            printf("SIM\n");
        else
            printf("NAO\n");

    } while (TesteFim(palavra) != 1); //testa se a palavra digitada foi "FIM", para assim parar a repetição
    return 0;
}

int testaPalindromo(char palavra[]) // função para testar se a palavra é um palindromo
{
    int i;
    int Nreal;
    Nreal = strlen(palavra);
    for (int i = 0; i < Nreal; i++)
    {
        if (palavra[i] != palavra[Nreal - 1 - i])
        {
            return 0;
        }
    }
    return 1;
}

int TesteFim(char palavra[])
{ // testa se a próxima palavra é fim
    int result = 0;
    if (strcmp(palavra, "FIM") == 0)
        result = 1;
    return result;
}

File: TP01/Palindromo.java
public class Palindromo {

  public static boolean isFim(String s) {
    return (s.length() == 3 && s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M');
  }

  public static void main(String[] args) {
    String palavra = "";
    String invertida = "";
    do {     // Testa se a palavra digitada foi "FIM", para assim parar a repetição
      palavra = MyIO.readLine();

      for (int i = (palavra.length() - 1); i >= 0; i--) { // Encontra o comprimento da String depois a inverte
        invertida += palavra.charAt(i);
      }
      if (invertida.equals(palavra)) { // Testa se a String contida na variavel nova é igual a da antiga
        MyIO.println("SIM");
      } else {
        MyIO.println("NAO");
      }
    } 
    while (!isFim(palavra));
  }
}
